### ISSUE TITLE
Organic groups access control not working for private access

### ISSUE THREAD

**Issue by schoenid** [backdrop-contrib/og#44]
The Group visibility field has been added to the group content type.
The group is enhanced by the field to select wether "private" or "public" access shall be granted.
After selecting "Private - accessible only to group members" and saving, the group is still accessible for all users, even users are not registered to that group.
A rebuild of the permissions (admin/reports/status/rebuild) does not solve the problem.

Backdrop 1.21.1

**Comment by schoenid** [comment]
After disabling the Content view access module, it is working fine.
So I guess, that permissions are overwritten ...

I need a solution, to prevent content visibility for unregistered users (guests).
As I see, it's really difficult to define appropriate rights for every case.
But Organic groups access control is working.

**Comment by olafgrabienski** [comment]
> After disabling the Content view access module, it is working fine.

Hm, I recall vaguely a module setting from Drupal 7 which allows to change the weight or priority of different content access modules used on the same site. Unfortunately, I don't remember which module provided the setting.

> I need a solution, to prevent content visibility for unregistered users (guests).

Have you tried to disable the permission `View published content` for the Anonymous role? If you want to differentiate between content types, you could try https://backdropcms.org/project/content_view_access.

**edit re Content View Access:** Apparently I didn't read your post thoroughly, as you already used that module, and it even caused the issue. Maybe post in that queue then?

**Comment by schoenid** [comment]
Thank you!
I found the module: Simple Access Grants.

> This does for now what I was looking for.
> It allows do define view access for registered users on a content type or node base.
> There is no conflict with Organic Groups. 

👎 NO.
After rebuilding permissions, I have the same problem.
To allow a view for registered users, overwrites the restriction, not to view private content. And it's not possible, to invert the permission. (No access for unregistered users ... this would solve the problem.)

I've tested also the Content Access module, which allows to define the content node grants priority between -10 ... 0 ... 10.
Without success.
So it seems to me, that I have to find a solution either on a per node basis, or by restricting URLs.
